deploy/001-psr.ts

```diff
@@ -19,19 +19,29 @@ const func: DeployFunction = async ({
   const acmAddress = (await ethers.getContractOrNull("AccessControlManager"))?.address || ADDRESS_ONE;
   const loopsLimit = 20;
 
+  const defaultProxyAdmin = await hre.artifacts.readArtifact(
+    "hardhat-deploy/solc_0.8/openzeppelin/proxy/transparent/ProxyAdmin.sol:ProxyAdmin",
+  );
+
   await deploy("ProtocolShareReserve", {
     from: deployer,
     log: true,
     deterministicDeployment: false,
     args: [comptrollerAddress, WBNBAddress, vBNBAddress],
+    skipIfAlreadyDeployed: true,
     proxy: {
       owner: live ? timelockAddress : deployer,
-      proxyContract: "OpenZeppelinTransparentProxy",
+      proxyContract: "OptimizedTransparentUpgradeableProxy",
       execute: {
         methodName: "initialize",
         args: [acmAddress, loopsLimit],
       },
+      viaAdminContract: {
+        name: "DefaultProxyAdmin",
+        artifact: defaultProxyAdmin,
+      },
     },
+    // maxFeePerGas: "200000000"  // Needed for zksync
```

Comment:
Could we set it conditionally considering the network?

Reply:
[ac7f495](https://github.com/VenusProtocol/protocol-reserve/pull/99/commits/ac7f49592b2d1795e464806534693510f940af0b)